exec/java-exec/src/test/java/org/apache/drill/exec/expr/fn/registry/FunctionRegistryHolderTest.java

```diff
@@ -155,6 +162,47 @@ public void testGetAllJarNames() {
     compareTwoLists(expectedResult, registryHolder.getAllJarNames());
   }
 
+  @Test
+  public void testGetAllJarsWithFunctionHolders() {
+    Map<String, List<FunctionHolder>> fnHoldersInRegistry = registryHolder.getAllJarsWithFunctionHolders();
+    //Iterate and confirm lists are same
+    for (String jarName : newJars.keySet()) {
+      List<DrillFuncHolder> expectedHolderList = newJars.get(jarName).stream()
+          .map(FunctionHolder::getHolder) //Extract DrillFuncHolder
+          .collect(Collectors.toList());
+      List<DrillFuncHolder> testHolderList = fnHoldersInRegistry.get(jarName).stream()
+          .map(FunctionHolder::getHolder) //Extract DrillFuncHolder
+          .collect(Collectors.toList());
+
+      compareTwoLists(expectedHolderList, testHolderList);
+    }
+
+    Map<String, String> shuffleFunctionMap = new HashMap<>();
+    // Confirm that same function spans multiple jars with different signatures
+    //Init: Expected Map of items
+    for (String jarName : newJars.keySet()) {
+      for (FunctionHolder funcHolder : newJars.get(jarName)) {
+        if (funcHolder.getName().equals(SHUFFLE_FUNC_NAME)) {
```

Comment:
General recommendation is not use equals on constant to avoid NPE: `SHUFFLE_FUNC_NAME.equals(funcHolder.getName())`. Please change here and in the code below.